### ISSUE TITLE
Floating point types of different precisions in padding not allowed

### ISSUE THREAD

**Issue by LSOFT123** [pytorch/xla#1761]
## 🐛 Bug

Not sure if the problem comes from Pytorch XLA itself or the EfficientNet model implementation but the same model trains fine with GPU. XRTCompile complains about float32 float64 mix in pytorch padding function which looks out of control of the EfficientNet model.

1. Install EfficientNet Pytorch: https://github.com/lukemelas/EfficientNet-PyTorch/
2. Instantiate a simple EfficientNet B3 model
3. Train it with 256x256 images

The problem seems to be in this **static_padding call in forward** function when resulting padding includes **some zeros in dimensions** :

```
class Conv2dStaticSamePadding(nn.Conv2d):
    """ 2D Convolutions like TensorFlow, for a fixed image size"""

    def __init__(self, in_channels, out_channels, kernel_size, image_size=None, **kwargs):
        super().__init__(in_channels, out_channels, kernel_size, **kwargs)
        self.stride = self.stride if len(self.stride) == 2 else [self.stride[0]] * 2

        # Calculate padding based on image size and save it
        assert image_size is not None
        ih, iw = image_size if type(image_size) == list else [image_size, image_size]
        kh, kw = self.weight.size()[-2:]
        sh, sw = self.stride
        oh, ow = math.ceil(ih / sh), math.ceil(iw / sw)
        pad_h = max((oh - 1) * self.stride[0] + (kh - 1) * self.dilation[0] + 1 - ih, 0)
        pad_w = max((ow - 1) * self.stride[1] + (kw - 1) * self.dilation[1] + 1 - iw, 0)
        if pad_h > 0 or pad_w > 0:
            self.static_padding = nn.ZeroPad2d((pad_w // 2, pad_w - pad_w // 2, pad_h // 2, pad_h - pad_h // 2))
        else:
            self.static_padding = Identity()

    def forward(self, x):
        x = self.static_padding(x)
        x = F.conv2d(x, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups)
        return x


class Identity(nn.Module):
    def __init__(self, ):
        super(Identity, self).__init__()

    def forward(self, input):
        return input
```

The error is:

```
Internal: From /job:tpu_worker/replica:0/task:0:
Seen floating point types of different precisions in %pad.1248 = f64[24,24,130,130]{3,2,1,0} pad(f32[24,24,128,128]{3,2,1,0} %get-tuple-element.1152, f64[] %constant.1247), padding=0_0x0_0x1_1x1_1, but mixed precision is disallowed.
	 [[{{node XRTCompile}}]]
```

This problem can be reproduced on Google Colab (installation by following your tutorial)
Python         : 3.6.9 (default, Nov  7 2019, 10:44:02) 
Numpy          : 1.17.5
PyTorch        : 1.5.0a0+e95282a
PyTorchXLA     : 0.8+58cc64c

**Comment by dlibenzi** [comment]
Yeah, it's a bug on our side. Will fix it soon.
Thanks for reporting!

**Comment by dlibenzi** [comment]
Oh, can you try using "nightly"?
Select "nightly" in the VERSION listbox in the environment configuration cell ...

**Comment by dlibenzi** [comment]
![image](https://user-images.githubusercontent.com/14933084/76707253-d3ac9d80-66aa-11ea-8f97-fd17b1634f39.png)

**Comment by LSOFT123** [comment]
Fixed in nightly. Thanks!
